xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -340,7 +335,13 @@ private Job startDeleteJob(EntityReference entityReference, XWikiContext context
         }
     }
 
-    protected DocumentReferenceResolver<String> getCurrentStringDocumentReferenceResolver()
+    /**
+     * @return the current document reference resolver to be used to resolve a string document into a
+     *         {@link org.xwiki.model.reference.DocumentReference}
+     * @since 14.4.1
+     * @since 14.5RC1
+     */
+    private DocumentReferenceResolver<String> getCurrentStringDocumentReferenceResolver()
```

Comment:
``DeleteAction`` is a component now, so it shouldn't rely on ``Utils.getComponent()`` anymore. Simply inject the required components.